boards/arm/stm32/emw3162/scripts/ld.script

```diff
@@ -45,11 +45,16 @@ SECTIONS
         *(.fixup)
         *(.gnu.warning)
 
-        . = ALIGN(0x4);
+        /* SDIO_RXDMA32_CONFIG and SDIO_TXDMA32_CONFIG values in stm32_sdio.c
+         * give the DMA burst length of 16 bytes
+         * (DMA_SCR_MSIZE_32BITS x DMA_SCR_MBURST_INCR4).
+         * Thus the following alignment should be 0x10.
+         */
+        . = ALIGN(0x10);
         wlan_firmware_image_location = .;
         *(.wlan_firmware_image .wlan_firmware_image.*)
         wlan_firmware_image_end = .;
-        . = ALIGN(0x4);
+        . = ALIGN(0x10);
```

Comment:
is it enough to add __attribute__((aligned(16))) on the global data?

Reply:
Ok, this is for WiFi mac firmware, not the network packet send to device, look good.

Reply:
Concerning firmware images, now I've removed ALIGN from the linker scripts, and added "aligned_data" to stm32_wlan_firmware.c instead.